faiss_vector_posting.go

```diff
@@ -475,7 +475,9 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, requiresFiltering bool
 							}
 						}
 
-						selector, err = faiss.NewIDSelectorNot(ineligibleVectorIDs)
+						if len(vectorIDsToInclude) > 0 {
```

Comment:
i think its better to short circuit these code paths to early exit the `rv` at 403 line and not have to do the downstream operations from thereon, since they're anyways not valid and we return an empty postings at 556.

Reply:
yeah, that makes it so much easier actually. great find!